File: gherkin/java/src/main/java/gherkin/pickles/PickleString.java
package gherkin.pickles;

public class PickleString implements Argument {
    private final PickleLocation location;
    private final String content;
    private final String contentType;

    public PickleString(PickleLocation location, String content, String contentType) {
        this.location = location;
        this.content = content;
        this.contentType = contentType;
    }

    public PickleString(PickleLocation location, String content) {
        this(location, content, null);
    }

    @Override
    public PickleLocation getLocation() {
        return location;
    }

    public String getContent() {
        return content;
    }

    public String getContentType() {
        return contentType;
    }
}

File: gherkin/dotnet/Gherkin/Pickles/PickleString.cs
using System.Linq;

namespace Gherkin.Pickles
{
    public class PickleString : Argument
    {
        private readonly PickleLocation location;
        public override PickleLocation Location
        {
            get { return location; }
        }

        public string Content { get; private set; }
        
        public string ContentType { get; private set; }

        public PickleString(PickleLocation location, string content, string contentType = null)
        {
            this.location = location;
            Content = content;
            ContentType = contentType;
        }
    }
}
